src/context/internal/uicontextresolver.cpp

```diff
@@ -51,6 +51,10 @@ void UiContextResolver::init()
             notation->interaction()->textEditingStarted().onNotify(this, [this]() {
                 notifyAboutContextChanged();
             });
+
+            notation->undoStack()->stackChanged().onNotify(this, [this]() {
+                notifyAboutContextChanged();
```

Comment:
This appears to be the most reliable trigger for ensuring all action's enabled status are updated as needed